src/main/java/net/nextencia/rrdiagram/BNFProcessor.java

```diff
@@ -0,0 +1,262 @@
+      logger.info("loaded: " + bnffile);
+      this.api = api;
+      this.version = version;
+    } catch (java.io.FileNotFoundException fne) {
+      logger.severe("File not found: " + bnffile);
+      System.exit(1);
+    } catch (java.io.IOException ioe) {
+      logger.severe("Unable to read from file: " + bnffile);
+      System.exit(1);
+    }
+  }
+
+  public String getReferenceFile() {
+    StringBuilder sb = new StringBuilder();
+
+    try {
+      checkGrammar();
+
+      GrammarToBNF bnf_builder = new GrammarToBNF();
+      GrammarToRRDiagram.RuleLinkProvider ruleLinkProvider =
+        new GrammarToRRDiagram.RuleLinkProvider() {
+          private String toLinkName(String name) {
+            return "../grammar_diagrams#" + name.replaceAll("_", "-");
+          }
+
+          @Override
+          public String getLink(String ruleName) {
+            return toLinkName(ruleName);
+          }
+        };
+
+      for (Rule rule : grammar.getRules()) {
+        try {
+          sb.append("### " + rule.getName() + "\n");
+
+          sb.append("```output.ebnf\n");
+          sb.append(rule.toYBNF());
+          sb.append("\n```\n");
+          GrammarToRRDiagram diagram_builder = new GrammarToRRDiagram();
+          diagram_builder.setRuleLinkProvider(ruleLinkProvider);
+          diagram_builder.setRuleConsideredAsLineBreak(Utils.lineBreakRule);
+
+          RRDiagram diagram = diagram_builder.convert(rule);
+          String svg_string = new RRDiagramToSVG().convert(diagram);
+          sb.append(svg_string);
+          sb.append("\n\n");
+        } catch (Exception e) {
+          logger.severe("Exception occurred while exporting rule " + rule.getName());
+          logger.warning(rule.toBNF(bnf_builder));
+          throw e;
+        }
+      }
+    } catch(Exception e) {
+      e.printStackTrace();
+    }
+    return sb.toString();
+  }
+
+  public String getGrammar(List<Rule> targetRules) {
+    StringBuilder sb = new StringBuilder();
+    try {
+      for (int i = 0; i < targetRules.size(); i++) {
+        if (i > 0) {
+          sb.append("\n");
+        }
+        sb.append(targetRules.get(i).toYBNF());
+        sb.append("\n");
+      }
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+    return sb.toString();
+  }
+
+  String getGlobalPrefix(int depth) {
+    /**
+     * This function returns the relative of the grammar_diagrams file to the current .md file
+     * reference file @ content/preview/api/ysql/syntax_resources/grammar_diagrams.md
+     * ysql/ycql is at depth 4 (from content)
+     */
+
+    String path = "";
+    
+    if (depth < 4) {
+      path = "/" + version + "/api/" + api + "/";
+    }
+
+    // for every level deeper from ysql/ycql, add a ../
+    switch(depth) {
+      case 5: path = "../"; break;
+      case 6: path = "../../"; break;
+      case 7: path = "../../../"; break;
+      case 8: path = "../../../../"; break;
+      case 9: path = "../../../../../"; break;
+    }
```

Comment:
it could be programmatically calculated, not case-by-case

Reply:
Yes, you are right. it is just faster this way.